### PR TITLE
feat: add subagent-prompt + write-plan skills, route review-frontend to remix-v2

### LINKED ISSUE #8: feat: add write-plan skill optimized for Claude Code
## Context

The `superpowers:write-plan` skill produces plans with large code examples that are never used when Claude Code is the executor. This is because the skill was designed for human engineers with zero codebase context who would copy-paste code.

### The Problem

The current skill (from superpowers, mirrored in `.gemini/commands/writing-plans.toml`) explicitly requires:

> "Complete code in plan (not 'add validation')" (line 108)
> "Write comprehensive implementation plans assuming the engineer has zero context for our codebase"

But **Claude Code doesn't work that way**:
1. It already has codebase context (can read any file)
2. It generates code dynamically from understanding, not by copying from plans
3. Complete code examples become dead weight - consuming tokens but never used
4. The executor regenerates everything anyway based on its own understanding

### Evidence

Observed in multiple plan executions:
- Plans contain 50-100+ lines of example code per task
- Executor never references the code examples - it reads actual files and generates fresh
- Token budget consumed by unused content
- Plans become harder to review due to noise

---

## Proposed Solution

Create `commands/write-plan.md` in beagle - a Claude Code-optimized version that:

1. **Replaces code examples with references** - "Follow the pattern in `src/auth/middleware.py:45-60`" instead of copying the code
2. **Focuses on intent and constraints** - What to build, what to avoid, edge cases
3. **Trusts the executor's codebase access** - No need to duplicate file contents
4. **Keeps interface signatures** - Type definitions and function signatures are useful, full implementations are not

---

## Task 1: Create the Skill

Create `commands/write-plan.md`:

```markdown
---
description: Create implementation plans optimized for Claude Code execution - focuses on intent, constraints, and references rather than code examples
---

# Writing Plans (Claude Code Optimized)

## Overview

Write implementation plans for Claude Code as executor. Unlike plans for human engineers, these assume the executor:
- Has full codebase access and can read any file
- Generates code dynamically from understanding
- Doesn't copy-paste from plans

**Announce at start:** "I'm using the write-plan skill to create the implementation plan."

**Save plans to:** `docs/plans/YYYY-MM-DD-<feature-name>.md`

## What to Include

### DO Include:
- **Intent**: What each task accomplishes and why
- **Constraints**: What NOT to do, edge cases, gotchas
- **References**: "Follow pattern in `file.py:L45-60`" (executor reads the file)
- **Interface signatures**: Type definitions, function signatures
- **Test criteria**: What behavior to verify
- **Dependencies**: Task ordering, what must exist first

### DON'T Include:
- Full code implementations (executor generates these)
- Code that duplicates existing patterns (just reference them)
- Content the executor will read from files anyway

## Plan Header

```markdown
# [Feature Name] Implementation Plan

**Goal:** [One sentence]

**Architecture:** [2-3 sentences on approach]

**Key Files:** 
- `path/to/main.py` - [what changes]
- `path/to/test.py` - [what to test]

---
```

## Task Structure

```markdown
### Task N: [Component Name]

**Files:**
- Create: `exact/path/to/file.py`
- Modify: `exact/path/to/existing.py` (the `function_name` function)
- Test: `tests/exact/path/to/test.py`

**Intent:** [What this accomplishes in 1-2 sentences]

**Approach:**
- Follow pattern in `src/similar/feature.py:45-60`
- Interface: `async def new_function(arg: Type) -> ReturnType`
- Must handle: [edge case 1], [edge case 2]
- Must NOT: [constraint]

**Test Criteria:**
- Verify [behavior 1]
- Verify [behavior 2]
- Edge case: [what happens when X]

**Commit:** `feat: add [component]`
```

## Anti-Patterns

❌ **Don't do this:**
```markdown
**Step 3: Write minimal implementation**
\`\`\`python
def function(input):
    validated = validate(input)
    result = process(validated)
    return format_output(result)
\`\`\`
```

✅ **Do this instead:**
```markdown
**Approach:**
- Follow validation pattern in `src/utils/validators.py:23-35`
- Interface: `def function(input: InputType) -> OutputType`
- Must validate input before processing
- Must format output consistently with other endpoints
```

## Why This Works

Claude Code:
1. Reads the referenced file when it needs the pattern
2. Understands the codebase conventions from exploration
3. Generates code that fits the actual current state of files
4. Doesn't drift from outdated examples in plans

---

## Execution Handoff

After saving the plan:

**"Plan saved to `docs/plans/<filename>.md`. Ready to execute with superpowers:execute-plan or task-by-task in this session."**
```

---

## Task 2: Add Skill Description

Ensure the skill appears in plugin manifest with description:

> "Create implementation plans optimized for Claude Code - uses references and intent rather than code examples"

---

## Task 3: Document the Difference

Add a note in beagle README or skill index explaining:

- `write-plan` (beagle) - For Claude Code execution, reference-based
- `write-plan` (superpowers) - For human engineers, complete code examples

---

## Success Criteria

- [ ] `commands/write-plan.md` created with reference-based approach
- [ ] Skill produces plans ~50% shorter than superpowers version
- [ ] Plans contain file references instead of duplicated code
- [ ] Plans remain actionable - executor can still complete tasks
- [ ] Anti-patterns documented to prevent regression

---

## Future Considerations

- Track plan execution success rate vs superpowers version
- Consider making this the default when executor is detected as Claude Code
- May want to add a "verbosity" flag for cases where more detail is needed

### PR DESCRIPTION
**Pull request by anderskev**
## Summary

Adds two paired workflow skills that form a planning→execution handoff (`beagle-analysis:write-plan` → `beagle-core:subagent-prompt`), and updates `beagle-react:review-frontend` to delegate to `beagle-remix-v2:review-remix-v2` when Remix v2 is detected in `package.json`.

### Added

- **`beagle-core:subagent-prompt`** — user-invocable skill that produces a self-contained orchestration prompt handing the current session's work off to a fresh session for sub-agent execution. Enforces per-task verification commands and a final integration check before the orchestrator can report success. `disable-model-invocation: true`, `user-invocable: true`.
- **`beagle-analysis:write-plan`** — turns a finalized `brainstorm-beagle` spec at `.beagle/concepts/<slug>/spec.md` into a bite-sized, TDD-driven implementation plan at `.beagle/concepts/<slug>/plan.md`. Reads the spec, scans `CLAUDE.md` conventions, designs file structure, decomposes work into 2–5 minute steps with exact paths/commands, self-reviews against the spec, then writes after explicit user approval. On success offers `beagle-core:subagent-prompt` as the execution handoff. Ported from the upstream osprey skill with refs rewritten to plugin-qualified format. Includes `references/plan-template.md` and `references/plan-reviewer.md`.

### Changed

- **`beagle-react:review-frontend`** had not been updated since `beagle-remix-v2` landed in #102. A React project that is actually a Remix v2 app was being reviewed against generic React Router patterns and missed all six Remix v2 area review skills (routing, data-flow, forms, error-boundaries, perf/SSR, meta+sessions).
  - Step 1 scope grep widened to include `.jsx`/`.mjs`/`.cjs` and `remix/vite` config files.
  - Step 2 adds an `@remix-run/*` detection grep against `package.json`.
  - Step 4 splits into Remix v2 vs non-Remix branches: when Remix v2 is detected, load the umbrella `beagle-remix-v2:review-remix-v2` and skip `beagle-react:react-router-code-review` (Remix v2 routes through its own route module conventions, which `remix-v2-routing-review` covers in context). Both branches still load `shadcn-code-review` and the orthogonal conditional skills (react-flow, zustand, tailwind-v4, vitest).
  - Gate 2 rewritten to enforce branch selection.
- **CHANGELOG.md** `[Unreleased]` Added entries for both new skills.
- **CLAUDE.md** skill counts: 143 → 145, beagle-core 19 → 20, beagle-analysis 12 → 13. New skills added to the Key Skills table.
- **plugin.json keywords:** `subagent-prompt`, `orchestration` in beagle-core; `write-plan`, `tdd-planning`, `implementation-plan` in beagle-analysis.

## Motivation

`write-plan` closes the gap between `brainstorm-beagle` (which produces a spec) and execution — previously there was no canonical, TDD-disciplined planner inside beagle, only ad-hoc plan documents. `subagent-prompt` closes the matching gap on the execution side: a portable handoff prompt that runs each task in its own sub-agent context with verification, instead of the current session burning through context window on implementation. The `review-frontend` change makes the existing comprehensive-review entry point actually load the Remix v2 review surface when present, instead of silently missing six areas of review coverage.

## Testing

No build, lint, or test suite — this is a pure markdown marketplace. Manual checks performed:

- `write-plan` SKILL.md is 394 lines (under the 500-line cap).
- `subagent-prompt` SKILL.md follows existing user-invocable workflow patterns (`commit-push`, `create-pr`) with `disable-model-invocation: true` + `user-invocable: true`.
- All cross-skill refs use plugin-qualified format (`beagle-analysis:brainstorm-beagle`, `beagle-core:subagent-prompt`) per `CLAUDE.md` conventions.
- `review-frontend` Step 2 detection grep is consistent with the patterns `review-remix-v2` already uses.
- `plugin.json` files remain valid JSON.

### Manual Testing Steps

1. In a Remix v2 project, invoke `/beagle-react:review-frontend` and confirm the umbrella `beagle-remix-v2:review-remix-v2` loads and `react-router-code-review` does not.
2. In a non-Remix React project, invoke `/beagle-react:review-frontend` and confirm the original behavior is preserved (loads `react-router-code-review`, no Remix v2 skills).
3. With a `.beagle/concepts/<slug>/spec.md` present, invoke `/beagle-analysis:write-plan` and confirm it locates the spec, drafts a plan, and gates on user approval before writing.
4. With work to hand off, invoke `/beagle-core:subagent-prompt` and confirm it produces a self-contained orchestration prompt that names paths, sub-agent types, and per-task verification commands.

## Version

No version bump — release flow handles `marketplace.json` and per-plugin `plugin.json` version bumps per `CLAUDE.md` §Release Process.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: plugins/beagle-analysis/skills/write-plan/references/plan-reviewer.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+**Purpose:** Verify the plan is complete, matches the spec, and has proper task decomposition.
+
+**Dispatch when:** The plan is long (>10 tasks), touches unfamiliar territory, or has high stakes. For short, familiar plans, skip the dispatched review — the self-review and the human review step are enough.
+
+**Dispatch after:** The complete plan draft is written and self-review has been honestly passed. Do not dispatch the reviewer to do your self-review for you.
+
+## Prompt Template
+
+```
```
[the remark this answers is not in the document]
Comment (anderskev) [reply plugins/beagle-analysis/skills/write-plan/references/plan-reviewer.md]: Fixed in 41e9307 — added `text` language tag to the fenced block.

File: plugins/beagle-analysis/skills/write-plan/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,394 @@
+---
+name: write-plan
+description: "Use when you have a finalized `beagle-analysis:brainstorm-beagle` spec at `.beagle/concepts/<slug>/spec.md` and need a bite-sized, TDD-driven implementation plan before any code is written. Triggers on: \"write a plan\", \"plan this spec\", \"turn the spec into a plan\", \"now plan the implementation\", \"/write-plan\". Reads the spec, designs the file structure, decomposes work into 2-5 minute TDD steps with exact paths and commands, self-reviews against the spec, gets user approval, then writes to `.beagle/concepts/<slug>/plan.md`. Does NOT brainstorm specs, write code, or execute the plan — produces the plan document only. Make sure to use this skill whenever a spec exists and the user is about to start building, even if they don't explicitly say \"plan\"."
+---
```
[the remark this answers is not in the document]
Comment (anderskev) [reply plugins/beagle-analysis/skills/write-plan/SKILL.md:5]: Fixed in 41e9307. Added `disable-model-invocation: true` and also removed the trailing "Make sure to use this skill whenever a spec exists…" sentence from the description since it was auto-invocation copy that the gate would silently neutralize.

File: plugins/beagle-analysis/skills/write-plan/SKILL.md [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+3. **Read project conventions** — scan `CLAUDE.md` (root and nested) for testing, commenting, and architecture rules the plan must respect
+4. **Explore relevant code** — read existing files the plan will touch or pattern-match against; do not guess at structure
+5. **Design file structure** — map which files will be created or modified before any task is written
+6. **Decompose into tasks** — each task is bite-sized (2-5 minute steps), TDD-driven, with exact paths and code
+7. **Self-review** — check against the spec, scan for placeholders, verify type consistency (see *Self-Review* below)
+8. **Optionally dispatch a reviewer subagent** — only for plans that are long or cover unfamiliar territory (see `references/plan-reviewer.md`)
+9. **Present draft to user** — show the draft in chat for review; iterate if needed
+10. **Write to disk** — save to `.beagle/concepts/<slug>/plan.md` only after explicit user approval
+
+```
```
[the remark this answers is not in the document]
Comment (anderskev) [reply plugins/beagle-analysis/skills/write-plan/SKILL.md]: Fixed in 41e9307 — added `text` language tag to the decision-tree fenced block.

File: plugins/beagle-analysis/skills/write-plan/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,394 @@
...
+2. **Path gate:** Target path is finalized — default `.beagle/concepts/<slug>/plan.md`, slug inherited from the spec's folder.
+
+- **Default path:** `.beagle/concepts/<slug>/plan.md`
+- **Slug source:** inherit from the spec's parent folder (the `<slug>` segment under `.beagle/concepts/`). User preferences override the default path.
+- Commit to git with message: `docs: add <slug> implementation plan`
+- After writing, tell the user:
+  > "Plan written to `<path>`. Review it on disk and let me know if you want changes, or hand it off to your executor."
+- Then ask whether the user wants a `beagle-core:subagent-prompt` to execute the plan in a new session — phrase it as a direct question so the user can answer yes/no:
+  > "Want me to run `beagle-core:subagent-prompt` so you can execute this in a fresh session?"
+- Wait for the next instruction before considering work complete.
```
[the remark this answers is not in the document]
Comment (anderskev) [reply plugins/beagle-analysis/skills/write-plan/SKILL.md]: Fixed in 41e9307 — auto-commit is now gated behind explicit user request (`If the user explicitly asks to commit, use: \`docs: add <slug> implementation plan\``).

[the remark this answers is not in the document]
Comment (anderskev) [reply plugins/beagle-analysis/skills/write-plan/SKILL.md]: Fixed in 41e9307. Verified that `subagent-prompt` SKILL.md does set `disable-model-invocation: true`. The handoff now instructs the user to invoke `/beagle-core:subagent-prompt` themselves in a fresh session and notes why.
